python/ray/util/client/server/dataservicer.py

```diff
@@ -27,17 +27,18 @@
 QUEUE_JOIN_SECONDS = 10
 
 
-def _get_reconnecting_from_context(context: Any) -> Optional[bool]:
+def _get_reconnecting_from_context(context: Any) -> bool:
     """
-    Get `reconnecting` from gRPC metadata, or None if missing.
+    Get `reconnecting` from gRPC metadata, or False if missing.
     """
     metadata = {k: v for k, v in context.invocation_metadata()}
     val = metadata.get("reconnecting")
     if val is None or val not in ("True", "False"):
         logger.error(
-            f'Client connecting with invalid value for "reconnecting": {val}')
-        context.set_code(grpc.StatusCode.FAILED_PRECONDITION)
```

Comment:
we don't need to context.set_code?

Reply:
If they're missing the metadata here then we should rely on the existing version mismatch check to end the connection, rather than setting an error in this stream